frontend/ui/elements/common_settings_menu_table.lua

```diff
@@ -541,24 +541,33 @@ common_settings.document = {
 local metadata_folder_str = {
     ["doc"] = _("book folder"),
     ["dir"] = "koreader/docsettings/",
+    ["hash"] = "koreader/hashdocsettings/"
 }
 
 local metadata_folder_help_text = _([[
 Book view settings, reading progress, highlights, bookmarks and notes (collectively known as metadata) are stored in a separate folder named <book-filename>.sdr (".sdr" meaning "sidecar").
 
 You can decide between two locations where these will be saved:
 - alongside the book file itself (the long time default): these sdr folders will be visible when you browse your library directories with another file browser or from your computer, which may clutter your vision of your library. But this allows you to move them along when you reorganize your library, and also survives any renaming of parent directories. Also, if you perform directory synchronization or backups, your settings will be part of them.
-- all inside koreader/docsettings/: these sdr folders will only be visible and used by KOReader, and won't clutter your vision of your library directories with another file browser or from your computer. But any reorganisation of your library (directories or filename moves and renamings) may result in KOReader not finding your previous settings for these books. These settings won't be part of any synchronization or backups of your library.]])
+- all inside koreader/docsettings/: these sdr folders will only be visible and used by KOReader, and won't clutter your vision of your library directories with another file browser or from your computer. But any reorganisation of your library (directories or filename moves and renamings) may result in KOReader not finding your previous settings for these books. These settings won't be part of any synchronization or backups of your library.
+- all inside koreader/hashdocsettings/ as hashes: sdr folders are identified not by filepath and filename but by partial MD5 hash. This allows you to rename, move, and copy documents at will and and KOReader will link them to their metadata. However, any file modifications such as writing highlights into PDFs or downloading from Calibre may change the hash, resulting in metadata being unlinked from the file. Calculating file hashes may also impact loading times.]])
```

Comment:
- double "and" in "at will and and KOReader".
- "This allows you to rename, move, and copy documents at will" - we can already do all of that from KOReader itself - You need to specify that "this allows you to .... FROM THE OUTSIDE" or something like that.
- "and KOReader will link them to their metadata" : a bit unclear to me (non native english speaker) what "link them" means. I guess it means KOReader will find them, but "link" evokes something like a hard or soft link to me, something physical :) But ok if it reads fine in English.
- "Calculating file hashes may also impact loading times." : and File browser browsing times.

This text states what it does, but doesn't really say why a user would want that. You may want to mention some (ie. yours :) use case.

Reply:
> a bit unclear to me (non native english speaker) what "link them" means. I guess it means KOReader will find them

'Link' reads naturally to me since it means 'connect' here. I couldn't think of a better word here but don't care much if you have a preference.

I added more use case info, although I think the ability to rename/move files outside KOReader will already be appealing to some.